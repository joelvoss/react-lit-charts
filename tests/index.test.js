import * as React from 'react';
import { render } from './test-utils';

import {
  Bars,
	Box,
	Chart,
	Columns,
	Grid,
	Point,
	Svg,
	SvgArea,
	SvgCircle,
	SvgLine,
	SvgPolygon,
	SvgRect,
	SvgScatterPlot,
} from '../src';

describe('Grid', () => {
	const Comp = ({ ticks, vertical }) => {
		return (
			<div style={{ width: 400, height: 300 }}>
				<Chart x1={0} x2={100} y1={0} y2={100}>
					<Grid vertical={vertical} count={5} ticks={ticks}>
						{({ value, first, last }) => {
							return (
								<div data-first={first} data-last={last}>
									{value}
								</div>
							);
						}}
					</Grid>
				</Chart>
			</div>
		);
	};

	it('should not have ARIA violations', async () => {
		const { container } = render(<Comp />);
		await expect(container).toHaveNoAxeViolations();
	});

	it('should render proper HTML', async () => {
		const { baseElement, rerender } = render(<Comp />);
		expect(baseElement).toMatchSnapshot();

		rerender(<Comp vertical />);
		expect(baseElement).toMatchSnapshot();
	});

	it('should render with custom ticks', async () => {
		const ticks = [0, 20, 50, 75];
		const { baseElement } = render(<Comp ticks={ticks} />);
		expect(baseElement).toMatchSnapshot();
	});
});

////////////////////////////////////////////////////////////////////////////////

describe('Svg', () => {
	const Comp = ({ clip }) => {
		return (
			<div style={{ width: 400, height: 300 }}>
				<Chart x1={0} x2={100} y1={0} y2={100}>
					<Svg clip={clip} />
				</Chart>
			</div>
		);
	};

	it('should not have ARIA violations', async () => {
		const { container } = render(<Comp />);
		await expect(container).toHaveNoAxeViolations();
	});

	it('should render proper HTML', async () => {
		const { baseElement, rerender } = render(<Comp />);
		expect(baseElement).toMatchSnapshot();

		rerender(<Comp clip />);
		expect(baseElement).toMatchSnapshot();
	});
});

////////////////////////////////////////////////////////////////////////////////

describe('SvgLine', () => {
	const defaultData = [
		{ x: 0, y: 0 },
		{ x: 20, y: 25 },
		{ x: 40, y: 10 },
		{ x: 60, y: 50 },
		{ x: 80, y: 75 },
		{ x: 100, y: 100 },
	];

	const Comp = ({ data = defaultData, x, y }) => {
		return (
			<div style={{ width: 400, height: 300 }}>
				<Chart x1={0} x2={100} y1={0} y2={100}>
					<Svg>
						<SvgLine data={data} x={x} y={y} />
					</Svg>
				</Chart>
			</div>
		);
	};

	it('should not have ARIA violations', async () => {
		const { container } = render(<Comp />);
		await expect(container).toHaveNoAxeViolations();
	});

	it('should render proper HTML', async () => {
		const { baseElement } = render(<Comp />);
		expect(baseElement).toMatchSnapshot();
	});

	it('should render custom data + accessors', async () => {
		const data = [
			{ date: 0, avg: 0 },
			{ date: 20, avg: 25 },
			{ date: 40, avg: 10 },
			{ date: 60, avg: 50 },
			{ date: 65, avg: 75 },
			{ date: 95, avg: 56 },
			{ date: 100, avg: 100 },
		];

		const x = d => d.date;
		const y = d => d.avg;

		const { baseElement } = render(<Comp data={data} x={x} y={y} />);
		expect(baseElement).toMatchSnapshot();
	});
});

////////////////////////////////////////////////////////////////////////////////

describe('SvgScatterPlot', () => {
	const defaultData = [
		{ x: 0, y: 0 },
		{ x: 20, y: 25 },
		{ x: 40, y: 10 },
		{ x: 60, y: 50 },
		{ x: 80, y: 75 },
		{ x: 100, y: 100 },
	];

	const Comp = ({ data = defaultData, x, y }) => {
		return (
			<div style={{ width: 400, height: 300 }}>
				<Chart x1={0} x2={100} y1={0} y2={100}>
					<Svg>
						<SvgScatterPlot data={data} x={x} y={y} />
					</Svg>
				</Chart>
			</div>
		);
	};

	it('should not have ARIA violations', async () => {
		const { container } = render(<Comp />);
		await expect(container).toHaveNoAxeViolations();
	});

	it('should render proper HTML', async () => {
		const { baseElement } = render(<Comp />);
		expect(baseElement).toMatchSnapshot();
	});

	it('should render custom data + accessors', async () => {
		const data = [
			{ date: 0, avg: 0 },
			{ date: 20, avg: 25 },
			{ date: 40, avg: 10 },
			{ date: 60, avg: 50 },
			{ date: 65, avg: 75 },
			{ date: 95, avg: 56 },
			{ date: 100, avg: 100 },
		];

		const x = d => d.date;
		const y = d => d.avg;

		const { baseElement } = render(<Comp data={data} x={x} y={y} />);
		expect(baseElement).toMatchSnapshot();
	});
});

////////////////////////////////////////////////////////////////////////////////

describe('SvgPolygon', () => {
	const defaultData = [
		{ x: 0, y: 0 },
		{ x: 20, y: 25 },
		{ x: 40, y: 10 },
		{ x: 60, y: 50 },
		{ x: 80, y: 75 },
		{ x: 100, y: 100 },
	];

	const Comp = ({ data = defaultData, x, y }) => {
		return (
			<div style={{ width: 400, height: 300 }}>
				<Chart x1={0} x2={100} y1={0} y2={100}>
					<Svg>
						<SvgPolygon data={data} x={x} y={y} />
					</Svg>
				</Chart>
			</div>
		);
	};

	it('should not have ARIA violations', async () => {
		const { container } = render(<Comp />);
		await expect(container).toHaveNoAxeViolations();
	});

	it('should render proper HTML', async () => {
		const { baseElement } = render(<Comp />);
		expect(baseElement).toMatchSnapshot();
	});

	it('should render custom data + accessors', async () => {
		const data = [
			{ date: 0, avg: 0 },
			{ date: 20, avg: 25 },
			{ date: 40, avg: 10 },
			{ date: 60, avg: 50 },
			{ date: 65, avg: 75 },
			{ date: 95, avg: 56 },
			{ date: 100, avg: 100 },
		];

		const x = d => d.date;
		const y = d => d.avg;

		const { baseElement } = render(<Comp data={data} x={x} y={y} />);
		expect(baseElement).toMatchSnapshot();
	});
});

////////////////////////////////////////////////////////////////////////////////

describe('SvgArea', () => {
	const defaultData = [
		{ x: 20, y: 25 },
		{ x: 40, y: 10 },
		{ x: 60, y: 50 },
		{ x: 80, y: 75 },
		{ x: 100, y: 100 },
	];

	const Comp = ({ data = defaultData, floor }) => {
		return (
			<div style={{ width: 400, height: 300 }}>
				<Chart x1={0} x2={100} y1={0} y2={100}>
					<Svg>
						<SvgArea data={data} floor={floor} />
					</Svg>
				</Chart>
			</div>
		);
	};

	it('should not have ARIA violations', async () => {
		const { container } = render(<Comp />);
		await expect(container).toHaveNoAxeViolations();
	});

	it('should render proper HTML', async () => {
		const { baseElement } = render(<Comp />);
		expect(baseElement).toMatchSnapshot();
	});

	it('should render with custom floor', async () => {
		const { baseElement } = render(<Comp floor={10} />);
		expect(baseElement).toMatchSnapshot();
	});
});

////////////////////////////////////////////////////////////////////////////////

describe('SvgCircle', () => {
	const Comp = ({ x = 50, y = 50 }) => {
		return (
			<div style={{ width: 400, height: 300 }}>
				<Chart x1={0} x2={100} y1={0} y2={100}>
					<Svg>
						<SvgCircle x={x} y={y} r={5} />
					</Svg>
				</Chart>
			</div>
		);
	};

	it('should not have ARIA violations', async () => {
		const { container } = render(<Comp />);
		await expect(container).toHaveNoAxeViolations();
	});

	it('should render proper HTML', async () => {
		const { baseElement } = render(<Comp />);
		expect(baseElement).toMatchSnapshot();
	});

	it('should render with custom x/y', async () => {
		const { baseElement } = render(<Comp x={20} y={0} />);
		expect(baseElement).toMatchSnapshot();
	});
});

////////////////////////////////////////////////////////////////////////////////

describe('SvgRect', () => {
	const Comp = ({ x1, x2, y1, y2 }) => {
		return (
			<div style={{ width: 400, height: 300 }}>
				<Chart x1={0} x2={100} y1={0} y2={100}>
					<Svg>
						<SvgRect x1={x1} x2={x2} y1={y1} y2={y2} />
					</Svg>
				</Chart>
			</div>
		);
	};

	it('should not have ARIA violations', async () => {
		const { container } = render(<Comp />);
		await expect(container).toHaveNoAxeViolations();
	});

	it('should render proper HTML', async () => {
		const { baseElement } = render(<Comp />);
		expect(baseElement).toMatchSnapshot();
	});

	it('should render with custom coordinates', async () => {
		const { baseElement } = render(<Comp x1={10} x2={20} y1={10} y2={20} />);
		expect(baseElement).toMatchSnapshot();
	});
});

////////////////////////////////////////////////////////////////////////////////

describe('Point', () => {
	const Comp = ({ x = 50, y = 50, children }) => {
		return (
			<div style={{ width: 400, height: 300 }}>
				<Chart x1={0} x2={100} y1={0} y2={100}>
					<Point x={x} y={y}>
						{children}
					</Point>
				</Chart>
			</div>
		);
	};

	it('should not have ARIA violations', async () => {
		const { container } = render(<Comp />);
		await expect(container).toHaveNoAxeViolations();
	});

	it('should render proper HTML', async () => {
		const { baseElement } = render(<Comp />);
		expect(baseElement).toMatchSnapshot();
	});

	it('should render with child', async () => {
		const { baseElement } = render(
			<Comp>
				<p>Child</p>
			</Comp>,
		);
		expect(baseElement).toMatchSnapshot();
	});
});

////////////////////////////////////////////////////////////////////////////////

describe('Box', () => {
	const Comp = ({ x1, x2, y1, y2, style }) => {
		return (
			<div style={{ width: 400, height: 300 }}>
				<Chart x1={0} x2={100} y1={0} y2={100}>
					<Box x1={x1} x2={x2} y1={y1} y2={y2} style={style} />
				</Chart>
			</div>
		);
	};

	it('should not have ARIA violations', async () => {
		const { container } = render(<Comp />);
		await expect(container).toHaveNoAxeViolations();
	});

	it('should render proper HTML', async () => {
		const { baseElement } = render(<Comp />);
		expect(baseElement).toMatchSnapshot();
	});

	it('should render with custom coordinates', async () => {
		const { baseElement } = render(
			<Comp
				x1={25}
				x2={75}
				y1={25}
				y2={75}
				style={{ backgroundColor: 'aquamarine' }}
			/>,
		);
		expect(baseElement).toMatchSnapshot();
	});
});

////////////////////////////////////////////////////////////////////////////////

describe('Columns', () => {
	const defaultData = [
		{ x: 20, y: 25 },
		{ x: 40, y: 10 },
		{ x: 60, y: 50 },
		{ x: 80, y: 75 },
		{ x: 100, y: 100 },
	];

	const Comp = ({ data = defaultData, x, y, width, children }) => {
		return (
			<div style={{ width: 400, height: 300 }}>
				<Chart x1={0} x2={100} y1={0} y2={100}>
					<Columns data={data} x={x} y={y} width={width}>
						{children}
					</Columns>
				</Chart>
			</div>
		);
	};

	it('should not have ARIA violations', async () => {
		const { container } = render(<Comp />);
		await expect(container).toHaveNoAxeViolations();
	});

	it('should render proper HTML', async () => {
		const { baseElement } = render(<Comp />);
		expect(baseElement).toMatchSnapshot();
	});

	it('should render with custom data + accessors', async () => {
		const data = [
			{ date: 20, avg: 25 },
			{ date: 40, avg: 10 },
			{ date: 60, avg: 50 },
			{ date: 65, avg: 75 },
			{ date: 95, avg: 56 },
			{ date: 100, avg: 100 },
		];

		const x = d => d.date;
		const y = d => d.avg;

		const { baseElement } = render(
			<Comp data={data} x={x} y={y} width={10}>
				<p>Child</p>
			</Comp>,
		);
		expect(baseElement).toMatchSnapshot();
	});
});

////////////////////////////////////////////////////////////////////////////////

describe('Bars', () => {
	const defaultData = [
		{ x: 20, y: 25 },
		{ x: 40, y: 10 },
		{ x: 60, y: 50 },
		{ x: 80, y: 75 },
		{ x: 100, y: 100 },
	];

	const Comp = ({ data = defaultData, x, y, height, children }) => {
		return (
			<div style={{ width: 400, height: 300 }}>
				<Chart x1={0} x2={100} y1={0} y2={100}>
					<Bars data={data} x={x} y={y} height={height}>
						{children}
					</Bars>
				</Chart>
			</div>
		);
	};

	it('should not have ARIA violations', async () => {
		const { container } = render(<Comp />);
		await expect(container).toHaveNoAxeViolations();
	});

	it('should render proper HTML', async () => {
		const { baseElement } = render(<Comp />);
		expect(baseElement).toMatchSnapshot();
	});

	it('should render with custom data + accessors', async () => {
		const data = [
			{ date: 20, avg: 25 },
			{ date: 40, avg: 10 },
			{ date: 60, avg: 50 },
			{ date: 65, avg: 75 },
			{ date: 95, avg: 56 },
			{ date: 100, avg: 100 },
		];

		const x = d => d.date;
		const y = d => d.avg;

		const { baseElement } = render(
			<Comp data={data} x={x} y={y} height={10}>
				<p>Child</p>
			</Comp>,
		);
		expect(baseElement).toMatchSnapshot();
	});
});
