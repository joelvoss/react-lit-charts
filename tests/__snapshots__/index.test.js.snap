// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bars should render proper HTML 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <div
          style="position: absolute; left: 0%; bottom: 24.5%; width: 20%; height: 1%;"
        />
        <div
          style="position: absolute; left: 0%; bottom: 9.5%; width: 40%; height: 1%;"
        />
        <div
          style="position: absolute; left: 0%; bottom: 49.5%; width: 60%; height: 1%;"
        />
        <div
          style="position: absolute; left: 0%; bottom: 74.5%; width: 80%; height: 1%;"
        />
        <div
          style="position: absolute; left: 0%; bottom: 99.5%; width: 100%; height: 1%;"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Bars should render with custom data + accessors 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <div
          style="position: absolute; left: 0%; bottom: 20%; width: 20%; height: 10%;"
        >
          <p>
            Child
          </p>
        </div>
        <div
          style="position: absolute; left: 0%; bottom: 5%; width: 40%; height: 10%;"
        >
          <p>
            Child
          </p>
        </div>
        <div
          style="position: absolute; left: 0%; bottom: 45%; width: 60%; height: 10%;"
        >
          <p>
            Child
          </p>
        </div>
        <div
          style="position: absolute; left: 0%; bottom: 70%; width: 65%; height: 10%;"
        >
          <p>
            Child
          </p>
        </div>
        <div
          style="position: absolute; left: 0%; bottom: 51%; width: 95%; height: 10%;"
        >
          <p>
            Child
          </p>
        </div>
        <div
          style="position: absolute; left: 0%; bottom: 95%; width: 100%; height: 10%;"
        >
          <p>
            Child
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Box should render proper HTML 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <div
          style="position: absolute; left: 0%; bottom: 0%; width: 1%; height: 1%;"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Box should render with custom coordinates 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <div
          style="background-color: aquamarine; position: absolute; left: 25%; bottom: 25%; width: 50%; height: 50%;"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Columns should render proper HTML 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <div
          style="position: absolute; left: 19.5%; bottom: 0%; width: 1%; height: 25%;"
        />
        <div
          style="position: absolute; left: 39.5%; bottom: 0%; width: 1%; height: 10%;"
        />
        <div
          style="position: absolute; left: 59.5%; bottom: 0%; width: 1%; height: 50%;"
        />
        <div
          style="position: absolute; left: 79.5%; bottom: 0%; width: 1%; height: 75%;"
        />
        <div
          style="position: absolute; left: 99.5%; bottom: 0%; width: 1%; height: 100%;"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Columns should render with custom data + accessors 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <div
          style="position: absolute; left: 15%; bottom: 0%; width: 10%; height: 25%;"
        >
          <p>
            Child
          </p>
        </div>
        <div
          style="position: absolute; left: 35%; bottom: 0%; width: 10%; height: 10%;"
        >
          <p>
            Child
          </p>
        </div>
        <div
          style="position: absolute; left: 55%; bottom: 0%; width: 10%; height: 50%;"
        >
          <p>
            Child
          </p>
        </div>
        <div
          style="position: absolute; left: 60%; bottom: 0%; width: 10%; height: 75%;"
        >
          <p>
            Child
          </p>
        </div>
        <div
          style="position: absolute; left: 90%; bottom: 0%; width: 10%; height: 56%;"
        >
          <p>
            Child
          </p>
        </div>
        <div
          style="position: absolute; left: 95%; bottom: 0%; width: 10%; height: 100%;"
        >
          <p>
            Child
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Grid should render proper HTML 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <div>
          <div
            style="position: absolute; left: 0px; top: 100%; width: 100%; height: 0px;"
          >
            <div
              data-first="true"
              data-last="false"
            >
              0
            </div>
          </div>
          <div
            style="position: absolute; left: 0px; top: 80%; width: 100%; height: 0px;"
          >
            <div
              data-first="false"
              data-last="false"
            >
              20
            </div>
          </div>
          <div
            style="position: absolute; left: 0px; top: 60%; width: 100%; height: 0px;"
          >
            <div
              data-first="false"
              data-last="false"
            >
              40
            </div>
          </div>
          <div
            style="position: absolute; left: 0px; top: 40%; width: 100%; height: 0px;"
          >
            <div
              data-first="false"
              data-last="false"
            >
              60
            </div>
          </div>
          <div
            style="position: absolute; left: 0px; top: 20%; width: 100%; height: 0px;"
          >
            <div
              data-first="false"
              data-last="false"
            >
              80
            </div>
          </div>
          <div
            style="position: absolute; left: 0px; top: 0%; width: 100%; height: 0px;"
          >
            <div
              data-first="false"
              data-last="true"
            >
              100
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Grid should render proper HTML 2`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <div>
          <div
            style="position: absolute; left: 0%; top: 0px; width: 0px; height: 100%;"
          >
            <div
              data-first="true"
              data-last="false"
            >
              0
            </div>
          </div>
          <div
            style="position: absolute; left: 20%; top: 0px; width: 0px; height: 100%;"
          >
            <div
              data-first="false"
              data-last="false"
            >
              20
            </div>
          </div>
          <div
            style="position: absolute; left: 40%; top: 0px; width: 0px; height: 100%;"
          >
            <div
              data-first="false"
              data-last="false"
            >
              40
            </div>
          </div>
          <div
            style="position: absolute; left: 60%; top: 0px; width: 0px; height: 100%;"
          >
            <div
              data-first="false"
              data-last="false"
            >
              60
            </div>
          </div>
          <div
            style="position: absolute; left: 80%; top: 0px; width: 0px; height: 100%;"
          >
            <div
              data-first="false"
              data-last="false"
            >
              80
            </div>
          </div>
          <div
            style="position: absolute; left: 100%; top: 0px; width: 0px; height: 100%;"
          >
            <div
              data-first="false"
              data-last="true"
            >
              100
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Grid should render with custom ticks 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <div>
          <div
            style="position: absolute; left: 0px; top: 100%; width: 100%; height: 0px;"
          >
            <div
              data-first="true"
              data-last="false"
            >
              0
            </div>
          </div>
          <div
            style="position: absolute; left: 0px; top: 80%; width: 100%; height: 0px;"
          >
            <div
              data-first="false"
              data-last="false"
            >
              20
            </div>
          </div>
          <div
            style="position: absolute; left: 0px; top: 50%; width: 100%; height: 0px;"
          >
            <div
              data-first="false"
              data-last="false"
            >
              50
            </div>
          </div>
          <div
            style="position: absolute; left: 0px; top: 25%; width: 100%; height: 0px;"
          >
            <div
              data-first="false"
              data-last="true"
            >
              75
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Point should render proper HTML 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <div
          style="position: absolute; width: 0px; height: 0px; left: 50%; top: 50%;"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Point should render with child 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <div
          style="position: absolute; width: 0px; height: 0px; left: 50%; top: 50%;"
        >
          <p>
            Child
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Svg should render proper HTML 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <svg
          preserveAspectRatio="none"
          style="position: absolute; width: 100%; height: 100%; overflow: visible;"
          viewBox="0 0 100 100"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Svg should render proper HTML 2`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <svg
          preserveAspectRatio="none"
          style="position: absolute; width: 100%; height: 100%; overflow: hidden;"
          viewBox="0 0 100 100"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`SvgArea should render proper HTML 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <svg
          preserveAspectRatio="none"
          style="position: absolute; width: 100%; height: 100%; overflow: visible;"
          viewBox="0 0 100 100"
        >
          <path
            d="M20,100L20,75L40,90L60,50L80,25L100,0L100,100"
            style="vector-effect: non-scaling-stroke;"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;

exports[`SvgArea should render with custom floor 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <svg
          preserveAspectRatio="none"
          style="position: absolute; width: 100%; height: 100%; overflow: visible;"
          viewBox="0 0 100 100"
        >
          <path
            d="M20,90L20,75L40,90L60,50L80,25L100,0L100,90"
            style="vector-effect: non-scaling-stroke;"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;

exports[`SvgCircle should render proper HTML 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <svg
          preserveAspectRatio="none"
          style="position: absolute; width: 100%; height: 100%; overflow: visible;"
          viewBox="0 0 100 100"
        >
          <circle
            cx="50"
            cy="50"
            r="5"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;

exports[`SvgCircle should render with custom x/y 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <svg
          preserveAspectRatio="none"
          style="position: absolute; width: 100%; height: 100%; overflow: visible;"
          viewBox="0 0 100 100"
        >
          <circle
            cx="20"
            cy="100"
            r="5"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;

exports[`SvgLine should render custom data + accessors 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <svg
          preserveAspectRatio="none"
          style="position: absolute; width: 100%; height: 100%; overflow: visible;"
          viewBox="0 0 100 100"
        >
          <path
            d="M0,100L20,75L40,90L60,50L65,25L95,44L100,0"
            style="vector-effect: non-scaling-stroke;"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;

exports[`SvgLine should render proper HTML 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <svg
          preserveAspectRatio="none"
          style="position: absolute; width: 100%; height: 100%; overflow: visible;"
          viewBox="0 0 100 100"
        >
          <path
            d="M0,100L20,75L40,90L60,50L80,25L100,0"
            style="vector-effect: non-scaling-stroke;"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;

exports[`SvgPolygon should render custom data + accessors 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <svg
          preserveAspectRatio="none"
          style="position: absolute; width: 100%; height: 100%; overflow: visible;"
          viewBox="0 0 100 100"
        >
          <path
            d="M0,100L20,75L40,90L60,50L65,25L95,44L100,0"
            style="vector-effect: non-scaling-stroke;"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;

exports[`SvgPolygon should render proper HTML 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <svg
          preserveAspectRatio="none"
          style="position: absolute; width: 100%; height: 100%; overflow: visible;"
          viewBox="0 0 100 100"
        >
          <path
            d="M0,100L20,75L40,90L60,50L80,25L100,0"
            style="vector-effect: non-scaling-stroke;"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;

exports[`SvgRect should render proper HTML 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <svg
          preserveAspectRatio="none"
          style="position: absolute; width: 100%; height: 100%; overflow: visible;"
          viewBox="0 0 100 100"
        >
          <rect
            height="1"
            width="1"
            x="0"
            y="99"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;

exports[`SvgRect should render with custom coordinates 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <svg
          preserveAspectRatio="none"
          style="position: absolute; width: 100%; height: 100%; overflow: visible;"
          viewBox="0 0 100 100"
        >
          <rect
            height="10"
            width="10"
            x="10"
            y="80"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;

exports[`SvgScatterPlot should render custom data + accessors 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <svg
          preserveAspectRatio="none"
          style="position: absolute; width: 100%; height: 100%; overflow: visible;"
          viewBox="0 0 100 100"
        >
          <path
            d="M0 100 A0 0 0 0 1 0.0001 100.0001 M20 75 A0 0 0 0 1 20.0001 75.0001 M40 90 A0 0 0 0 1 40.0001 90.0001 M60 50 A0 0 0 0 1 60.0001 50.0001 M65 25 A0 0 0 0 1 65.0001 25.0001 M95 44 A0 0 0 0 1 95.0001 44.0001 M100 0 A0 0 0 0 1 100.0001 0.0001"
            style="vector-effect: non-scaling-stroke;"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;

exports[`SvgScatterPlot should render proper HTML 1`] = `
<body>
  <div>
    <div
      style="width: 400px; height: 300px;"
    >
      <div
        style="position: relative; display: block; width: 100%; height: 100%;"
      >
        <svg
          preserveAspectRatio="none"
          style="position: absolute; width: 100%; height: 100%; overflow: visible;"
          viewBox="0 0 100 100"
        >
          <path
            d="M0 100 A0 0 0 0 1 0.0001 100.0001 M20 75 A0 0 0 0 1 20.0001 75.0001 M40 90 A0 0 0 0 1 40.0001 90.0001 M60 50 A0 0 0 0 1 60.0001 50.0001 M80 25 A0 0 0 0 1 80.0001 25.0001 M100 0 A0 0 0 0 1 100.0001 0.0001"
            style="vector-effect: non-scaling-stroke;"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;
